rmf_robot_sim_common/include/rmf_robot_sim_common/slotcar_common.hpp

```diff
@@ -167,7 +167,13 @@ class SlotcarCommon
 
   void publish_robot_state(const double time);
 
-  bool is_holonomic();
+  // steering type constants
+  static const std::string DIFF_DRIVE;
+  static const std::string ACKERMANN;
```

Comment:
I think we can use a hash map that maps a string to an `enum` (maybe called VehicleDriveType or something along those lines), so it would be a bit clearer, i.e. go from `SlotcarCommon::DIFF_DRIVE` to `VehicleDriveType::DIFF_DRIVE`.
Something similar to what we do for checking the physics engine plugin [here](https://github.com/open-rmf/rmf_simulation/blob/main/rmf_robot_sim_ignition_plugins/src/slotcar.cpp#L29-L31)

Reply:
switch to enum 2e157ce, no need for the unordered map atm since it's only used in a single area

Reply:
I did a quick grep in the rmf codebase and noticed that we don't do uppercase `enum class`, the convention seems to be on `UpperCamelCase`, also having to write `SlotcarCommon::STEERING_TYPE::ACKERMANN` is quite verbose, you can bring the definition of the enum class outside of the `SlotcarCommon` class to get rid of the first scope. Since it would be in the `rmf_robot_sim_common` namespace there is no risk of polluting other namespaces.

Reply:
fixed 9e0001d